### DISCUSSION TITLE
Slow dashboard picker

### DISCUSSION THREAD

**Discussion by 22ispencer** [folke/snacks.nvim#294] [Q&A]
I found snacks a few days ago and decided it'd be fun to rewrite my conf again. Anyways now I'm back to using lazy.nvim instead of mini.deps, as such it's very possible that i have a bad config. My issue right now is i love the way the dashboard from snacks looks/works, but with my current config it feels like it takes at least 500ms for the picker to work once I open neovim. Below are my `lua/plugins/snacks.lua` and `lua/plugins/mini.lua`. Also I have tried making `"echasnovski/mini.pick"` a dependency of snacks and that doesn't seem to help much if at all. Any help figuring this out would be great! 

`lua/plugins/snacks.lua`
```lua
return {
	"folke/snacks.nvim",
	priority = 1000,
	lazy = false,
	opts = {
		-- your configuration comes here
		-- or leave it empty to use the default settings
		-- refer to the configuration section below
		bigfile = { enabled = true },
		dashboard = { enabled = true },
		indent = { enabled = true },
		input = { enabled = true },
		notifier = { enabled = true },
		quickfile = { enabled = true },
		scroll = { enabled = true },
		statuscolumn = { enabled = true },
		words = { enabled = true },
	},
}

```

`lua/plugins/mini.lua`
```lua
﻿return {
	{ "echasnovski/mini.ai", opts = {} },
	{ "echasnovski/mini.basics", opts = {} },
	{ "echasnovski/mini.bracketed", opts = {} },
	{ "echasnovski/mini.comment", opts = {} },
	{ "echasnovski/mini.completion", opts = {} },
	{ "echasnovski/mini.cursorword", opts = {} },
	{ "echasnovski/mini.diff", opts = {} },
	{ "echasnovski/mini.extra", opts = {} },
	{
		"echasnovski/mini.files",
		opts = {},
		keys = {
			{
				"<Leader>e",
				function()
					MiniFiles.open()
				end,
			},
		},
	},
	{ "echasnovski/mini.fuzzy", opts = {} },
	{ "echasnovski/mini-git", main = "mini.git", opts = {} },
	{ "echasnovski/mini.icons", opts = {} },
	{ "echasnovski/mini.jump", opts = {} },
	{ "echasnovski/mini.jump2d", opts = {} },
	{ "echasnovski/mini.pairs", opts = {} },
	{
		"echasnovski/mini.pick",
		config = function()
			require("mini.pick").setup({})
			vim.ui.select = MiniPick.ui_select
		end,
		keys = {
			{ "<Leader><Leader>", "<cmd>Pick buffers<cr>" },
			{ "<Leader>ff", "<cmd>Pick files<cr>" },
			{ "<Leader>fh", "<cmd>Pick help<cr>" },
			{ "<Leader>fg", "<cmd>Pick grep_live<cr>" },
		},
	},
	{ "echasnovski/mini.sessions", opts = {} },
	{ "echasnovski/mini.splitjoin", opts = {} },
	{ "echasnovski/mini.statusline", opts = {} },
	{ "echasnovski/mini.surround", opts = {} },
	{ "echasnovski/mini.tabline", opts = {} },
}
```

**Comment by folke** [comment]
This really has nothing to do with snacks. If mini.pick is slow for some reason, the ask in that repo.

**Comment by jcomesana** [comment]
I had this issue too and it took me a while to understand it. In my case, pressing "f" to select files with fzf-lua was fast, but pressing "g" to search text had a couple of seconds of delay. 

The problem was not that the picker was slow, it was a combination of the chosen key (that's "g") and having a big value for `vim.o.timeoutlen`. In neovim or vim, g is a key that is used in combination with other keys (gf, gu, gU, ...) so if you press only "g", vim/neovim will wait for `timeoutlen` milliseconds for the next keys.

My setting for `timeoutlen` is 2000, so here is my delay.

Possible solutions, set another key (I've chosen "t" instead of "g") or set a smaller value for `timeoutlen`.
